src/main/java/com/artozersky/HackerNewsAPI/service/impl/PostServiceImpl.java

```diff
@@ -26,58 +34,128 @@ public class PostServiceImpl implements NewsPostService {
     @Autowired
     private ModelMapper modelMapper;
 
+    // @Autowired
+    private SimpleCache cacheService;
+
     @Autowired
-    private CacheServiceImpl<Long, NewsPostModel> cacheService;
+    public PostServiceImpl(@Value("${cache.size:100}") int cacheSize) {
+        this.cacheService = new SimpleCache(cacheSize);
+    }
     
     @Override
-    public PostResponseDTO deletePost(Long postId) {
-        
-        NewsPostModel post = postRepository.findById(postId)
+    public  PostResponseDTO getPostById(Long postId) {
+        try{
+            // Step 1: Try to get the NewsPostModel from the cache
+            NewsPostModel cachedValue = cacheService.get(postId);
+            if (cachedValue != null) {
```

Comment:
this is what i mean to do a cache service, your service functions need to be short and simple. so instead of handling cache logic here do it in the cache service abstraction so your service layer will focus on business only not cache logic to validate things.

Reply:
fixed